specification/validated-relationships-service-api.yaml

```diff
@@ -1089,119 +1156,97 @@ components:
           type: number
           description: |
             The number of resources contained within the Bundle.
-
-            An empty bundle suggests that the proxy does not have patients they can act on behalf of.
           example: 2
-        link:
-          type: array
-          items:
-            type: object
+
+    Searchset:
+      allOf:
+          - $ref: "#/components/schemas/BaseBundle"
+          - type: object                       
             properties:
-              relation:
-                type: string
-                description: |
-                  Links related to this Bundle - see:
-                  http://www.iana.org/assignments/link-relations/link-relations.xhtml
-                example: "self"
-              url:
-                type: string
-                format: uri
-                example: https://api.service.nhs.uk/validated-relationships/FHIR/R4/RelatedPerson?_include=RelatedPerson%3apatient&identifier=https%3A%2F%2Ffhir.nhs.uk%2FId%2Fnhs-number%7C9000000017
-        entry:
-          type: array
-          description: |
-            A collection of resources contained within the Bundle.
+              type:
+                  type: string
+                  description: Denotes that the Bundle is a collection of resources returned as a result of a search.
+                  enum: [searchset]
+     
+    RelatedPersonBundle:
+      allOf:
+        - $ref: "#/components/schemas/Searchset"
+        - type: object          
+          properties:
+            link:
+              type: array
+              items:
+                type: object
+                properties:
+                  relation:
+                    type: string
+                    description: |
+                      Links related to this Bundle - see:
+                      http://www.iana.org/assignments/link-relations/link-relations.xhtml
+                    example: "self"
+                  url:
+                    type: string
+                    format: uri
+                    example: https://api.service.nhs.uk/validated-relationships/FHIR/R4/RelatedPerson?_include=RelatedPerson%3apatient&identifier=https%3A%2F%2Ffhir.nhs.uk%2FId%2Fnhs-number%7C9000000017
+            entry:
+              type: array
+              description: |
+                A collection of resources contained within the Bundle.
 
-            An empty bundle suggests that the proxy does not have patients they can act on behalf of.
-          items:
-            anyOf:
-              - $ref: "#/components/schemas/RelatedPerson"
-              - $ref: "#/components/schemas/Patient"
+                An empty bundle suggests that no candidate proxy/patient links were identified.
+              items:
+                anyOf:
+                  - $ref: "#/components/schemas/RelatedPersonBundleEntry"
+                  - $ref: "#/components/schemas/PatientBundleEntry"
+
+    ConsentBundle:
+      allOf:
```

Comment:
Do you need to specify "allOf" at this level? Polymorphism does give my brain a wedgie, so it's possible that I'm missing something.

Reply:
Yes, to enforce the shape of the response matches the composite of the base searchset and the `entry` relevant for a consent searchset bundle - effectively we're extending the searchset base type